src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java

```diff
@@ -281,5 +291,83 @@ private void updateRequestStatus(AssignmentRequest assignmentRequest, Status sta
 
     }
 
+    @SuppressWarnings("unchecked")
+    public ResponseEntity<Void> deleteRoleAssignmentByQuery(MultipleQueryRequest multipleQueryRequest) {
+
+        //1. create the request Object
+        if (CollectionUtils.isNotEmpty(multipleQueryRequest.getQueryRequests())) {
```

Comment:
we need more validations here : 
1) The collection must not be empty - done
2) Every query item of the collection need to be very restrictive: 
  I think, CCD must send the list of actorIds otherwise this could potentially try to delete all the 
  selected records. I will get more details from CCD team and Jon.
3) The request must be originated from CCD. Maybe we can implement the role based access on this endpoint as it is very risky if accessible to other consumers.

Reply:
This was discussed with Jon and we don't need to do any extra validations for this.